frontend/src/components/labwork/step/LabworkStep.tsx

```diff
@@ -76,8 +76,8 @@ const LabworkStep = ({ protocol, step, stepSamples }: LabworkStepPageProps) => {
 	// ** Template handling **
 
 	// A selected template picker is used if protocol supports more than one template
-	const [selectedTemplate, setSelectedTemplate] = useState<LabworkPrefilledTemplateDescriptor>()
-
+	const [selectedTemplate, setSelectedTemplate] = useState<LabworkPrefilledTemplateDescriptor>();
+	const [selectAllSamples, setSelectAllSamples] = useState(false);
```

Comment:
I'm having a hard time understanding the `selectAllSamples` logic. For a boolean state, I would use a name like `canSelectAllSamples`. I would also expect it to be true if the button should be enabled, and false if it should be disabled. I know that it's used to set the `disabled` property on the button, but I think it would be better to do `disabled={!canSelectAllSamples}` than to have a flag that has the opposite value from what the casual reader would expect.

I think the only time this button should be disabled is if there are zero samples in the `displayedSamples` state. That should indicate that zero samples match the user's current set of filters, so selecting all samples would result in zero samples from the backend.